hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/WritableName.java

```diff
@@ -92,7 +92,7 @@ public static synchronized Class<?> getClass(String name, Configuration conf
                                             ) throws IOException {
     Class<?> writableClass = NAME_TO_CLASS.get(name);
     if (writableClass != null)
-      return writableClass.asSubclass(Writable.class);
+      return writableClass;
```

Comment:
It is a public method and you are changing the behaviour of this. Earlier for which classes it used to return an Exception.
Now it will be returning some result.

I would suggest add a new method, which can take an argument based on which it can do .asSubclass(Writable.class) or not, And you can refactor whatever is inside this method to that new method with an extra flag, and call it from the existing method with false or so.

Reply:
I was thinking this would be a safe change from a compatibility perspective. People probably shouldn't have code calling this with writables, since it would just fail. I would think the inverse (throwing a new exception) would be more problematic. Is there a documented compatibility guideline for Hadoop?

But yea I can do that if you think it's necessary, just might be nicer without.

Reply:
That is what we expect, nobody should be expecting an exception. But this method used to throw an exception earlier and we should ensure unless it is a bug in the code, or the last option we should let it stay as is. Such changes looks safe on hadoop side, but when a downstream project tries to upgrade and if it is using such methods, it creates issues and blocks the way for them.(Have seen that in the past with Hive/Tez)

So, I would say if we can avoid changing the existing behaviour of a public method, we should do as a best effort

Reply:
Sounds good @ayushtkn. This class is annotated:

```
@InterfaceAudience.Private
@InterfaceStability.Evolving
```

So I thought we had more leeway (though i'm more used to hbase project).

I just pushed the change you requested.

Reply:
thx. the method is there since 2009, that is deemed stable :)